### PR TITLE
Add documentation for the gpubsub producer

### PR DESCRIPTION
**Pull request by Agares**
Signed-off-by: Ramona Luczkiewicz <rluczkiewicz@wayfair.com>

### CODE REVIEW

File: docs/reference/connectors/gpubsub_producer.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+---
+sidebar_label: gpubsub_producer
```
Comment (Licenser) [inline docs/reference/connectors/gpubsub_producer.md:2]: Would it make sense to add a note that gpubsub is google pub sub as we do for gcs?
```suggestion
sidebar_label: gpubsub_producer (Google Pub/Sub)
```

**Comment by Licenser** [comment]
This needs some kind of explanation on the configuration parameters available right now a user couldn't make out what they can tweak and what the effect is.

**Review by Licenser — APPROVED** [review]
🚀 noice!!!!

File: docs/reference/connectors/gpubsub_producer.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
+---
+sidebar_label: gpubsub_producer (Google Pub/Sub)
+sidebar_position: 1
+---
+
+# The `gpubsub_consumer` Connector
```
Comment (mfelsche) [inline docs/reference/connectors/gpubsub_producer.md:6]:
```suggestion
# The `gpubsub_producer` Connector
```

File: docs/reference/connectors/gpubsub_producer.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+This connector allows producing to a Google PubSub queue.
+
+## Configuration
+
+The credentials must be provided in a JSON file. The path to the JSON file should be set as an environment variable called `GOOGLE_APPLICATION_CREDENTIALS`
```
Comment (mfelsche) [inline docs/reference/connectors/gpubsub_producer.md:12]: This should be updated to note that it also reads from wherever gcloud expects stuff and works magically when deployed on google machines

File: docs/reference/connectors/gpubsub_producer.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
```
Comment (mfelsche) [inline docs/reference/connectors/gpubsub_producer.md:13]: A table with the possible config options would.be nice

File: docs/reference/connectors/gpubsub_producer.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    with
+        config = {"interval": 1000}
+    end;
+
+    define connector gpub from gpubsub_producer
+    with
+        codec = "json",
+        config = {
+            "topic": "projects/xxx/topics/test-topic-a", # required - the identifier of the topic
+            "connect_timeout": 100000000, # optional - connection timeout (nanoseconds) - defaults to 10s
```
Comment (mfelsche) [inline docs/reference/connectors/gpubsub_producer.md:32]: Use the std::time::nanos::from_seconds function here

File: docs/reference/connectors/gpubsub_producer.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+    connect /connector/metro/out to /pipeline/passthrough;
+    connect /pipeline/passthrough to /connector/gpub/in;
+end;
+
+deploy flow gbqtest;
+```
+
+## Metadata
+The connector will use the `$pubsub_publisher` metadata variable, which can be used to set the `ordering_key`.
```
Comment (mfelsche) [inline docs/reference/connectors/gpubsub_producer.md:51]: The prefix of the metadata should be the same as the connector type: `gpubsub_producer` for consistency with other connectors.

**Review by darach — APPROVED** [review]
LGTM 👌

File: docs/reference/connectors/gpubsub_producer.md (hunk 2 above)
Comment (mfelsche) [reply docs/reference/connectors/gpubsub_producer.md:6]: This is still wrong :sob:

**Review by Licenser — DISMISSED** [review]

File: docs/reference/connectors/gpubsub_producer.md (hunk 2 above)
Comment (Agares) [reply docs/reference/connectors/gpubsub_producer.md:6]: ahh, sorry, my bad, fixing now :)

**Review by mfelsche — APPROVED** [review]

**Pull request merged**